From 2e6bf149908d61c2b2121dad36f2caef260a2053 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: May 28 2019 12:56:55 +0000
Subject: Keep FT_Outline_New_Internal() and FT_Outline_Done_Internal()
for ABI compatibility but make them just throw
Unimplemented_Feature error.

Remove them once soname has been bumped!

Resolves: #1689117

Conflict:NA
Reference:https://src.fedoraproject.org/rpms/freetype/c/2e6bf149908d61c2b2121dad36f2caef260a2053

--- freetype-2.10.0/include/freetype/ftoutln.h
+++ freetype-2.10.0/include/freetype/ftoutln.h
@@ -165,6 +165,15 @@ FT_BEGIN_HEADER
                   FT_Int       numContours,
                   FT_Outline  *anoutline );
 
+  /*
+   * Kept downstream for ABI compatibility only.
+   * It just throws error now. Remove once soname has been bumped.
+   */
+  FT_EXPORT( FT_Error )
+  FT_Outline_New_Internal( FT_Memory    memory,
+                           FT_UInt      numPoints,
+                           FT_Int       numContours,
+                           FT_Outline  *anoutline );
 
   /**************************************************************************
    *
@@ -192,6 +201,13 @@ FT_BEGIN_HEADER
   FT_Outline_Done( FT_Library   library,
                    FT_Outline*  outline );
 
+  /*
+   * Kept downstream for ABI compatibility only.
+   * It just throws error now. Remove once soname has been bumped.
+   */
+  FT_EXPORT( FT_Error )
+  FT_Outline_Done_Internal( FT_Memory    memory,
+                            FT_Outline*  outline );
 
   /**************************************************************************
    *
--- freetype-2.10.0/src/base/ftoutln.c
+++ freetype-2.10.0/src/base/ftoutln.c
@@ -291,6 +291,19 @@
 
   /* documentation is in ftoutln.h */
 
+  /*
+   * Kept downstream for ABI compatibility only.
+   * It just throws error now. Remove once soname has been bumped.
+   */
+  FT_EXPORT_DEF( FT_Error )
+  FT_Outline_New_Internal( FT_Memory    memory,
+                           FT_UInt      numPoints,
+                           FT_Int       numContours,
+                           FT_Outline  *anoutline )
+  {
+    return FT_THROW( Unimplemented_Feature );
+  }
+
   FT_EXPORT_DEF( FT_Error )
   FT_Outline_New( FT_Library   library,
                   FT_UInt      numPoints,
@@ -423,6 +436,17 @@
 
   /* documentation is in ftoutln.h */
 
+  /*
+   * Kept downstream for ABI compatibility only.
+   * It just throws error now. Remove once soname has been bumped.
+   */
+  FT_EXPORT_DEF( FT_Error )
+  FT_Outline_Done_Internal( FT_Memory    memory,
+                            FT_Outline*  outline )
+  {
+    return FT_THROW( Unimplemented_Feature );
+  }
+
   FT_EXPORT_DEF( FT_Error )
   FT_Outline_Done( FT_Library   library,
                    FT_Outline*  outline )
